sklearn/preprocessing/imputation.py

```diff
@@ -374,3 +379,482 @@ def transform(self, X):
             X[coordinates] = values
 
         return X
+
+
+class MICEImputer(BaseEstimator, TransformerMixin):
+    """MICE (Multivariate Imputations by Chained Equations) transformer to
+    impute missing values.
+
+    Basic implementation of MICE package from R. This version assumes all of
+    the features are Gaussian.
+
+    Parameters
+    ----------
+    missing_values : int or "NaN", optional (default="NaN")
+        The placeholder for the missing values. All occurrences of
+        `missing_values` will be imputed. For missing values encoded as
+        np.nan, use the string value "NaN".
+
+    imputation_order : str, optional (default="monotone")
+        The order in which the features will be imputed.
+        - "monotone" - From features with fewest missing values to most.
+        - "revmonotone" - From features with most missing values to fewest.
+        - "roman" - Left to right.
+        - "arabic" - Right to left.
+        - "random" - A random order for each round.
+
+    n_imputations : int, optional (default=100)
+        Number of MICE rounds to perform the results of which will be
+        used in the final average.
+
+    n_burn_in : int, optional (default=10)
+        Number of initial MICE rounds to perform the results of which
+        will not be returned.
+
+    estimator : object, default=BayesianRidgeRegression
+        The estimator to use at each step of the round-robin imputation.
+        As of now, it must support `return_std` in its ``predict`` method.
+
+    n_nearest_features : int, optional (default=None)
+        Number of other features to use to estimate the missing values of
+        the current features. Can provide significant speed-up when the number
+        of features is huge. If `None`, all features will be used.
+
+    initial_strategy : str, optional (default="mean")
+        Which strategy to use to initialize the missing values. Same as the
+        `strategy` option in :class:`sklearn.preprocessing.Imputer.`
+        Valid values: {"mean", "median", or "most_frequent"}.
+
+    min_value : float (default=None)
+        Minimum possible imputed value.
```

Comment:
I've seen some cases in practice where linear models extrapolate wildly in a way that doesn't make sense. If you happen to know that your features should never be out of some range, it can be useful. For example, if you pass a `MinMax` normalized `X`, you can stick in `0` and `1` for min and max. I can take it out, but it seemed like a nice, simple convenience feature.